From 2edd8db64706fba5b2796a4bef1083936c481d60 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 1 Dec 2015 14:42:41 +0800
Subject: [PATCH 0713/5242] MLK-11932 video: fbdev: mxc: ldb: Do not touch
 fbi->modelist in ldb_init()

commit  41b6fe4f166c65d1aef83d64ae7259c1f8270277 from
https://source.codeaurora.org/external/imx/linux-imx.git

We don't have to touch fbi->modelist in ldb_init(), because mxcfb_register()
may initialize the list and add a video mode to the list for us.

This patch fixes the following issue reported by Coverity:
	INIT_LIST_HEAD(&fbi->modelist);
Uninitialized scalar variable (UNINIT)
uninit_use_in_call: Using uninitialized value fb_vm. Field fb_vm.name is
uninitialized when calling fb_add_videomode.
	fb_add_videomode(&fb_vm, &fbi->modelist);
	fb_videomode_to_var(&fbi->var, &fb_vm);

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/ldb.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/video/fbdev/mxc/ldb.c b/drivers/video/fbdev/mxc/ldb.c
index e5716cb..1d6c9e0 100644
--- a/drivers/video/fbdev/mxc/ldb.c
+++ b/drivers/video/fbdev/mxc/ldb.c
@@ -323,9 +323,6 @@ static int ldb_init(struct mxc_dispdrv_handle *mddh,
 	chan->fbi = fbi;
 
 	fb_videomode_from_videomode(&chan->vm, &fb_vm);
-
-	INIT_LIST_HEAD(&fbi->modelist);
-	fb_add_videomode(&fb_vm, &fbi->modelist);
 	fb_videomode_to_var(&fbi->var, &fb_vm);
 
 	setting->crtc = chan->crtc;
-- 
1.7.9.5

